registry/src/metadata/postgresql.rs

```diff
@@ -160,36 +162,40 @@ impl Publish<PgsqlMetadataStorage> for PackageManifest {
             ));
         };
 
+        let mut tx = e.pool.begin().await.map_err(|x| {
+            error!("Error starting transaction: {}", x);
+            MetadataStorageError::Internal
+        })?;
+
         let version_hash = hash_string(self.version.to_string().as_str());
 
-        let pg_result = sqlx::query(INSERT_VERSION_QUERY)
+        let query = sqlx::query(INSERT_VERSION_QUERY)
             .bind(db_package.id)
             .bind(self.version.to_string())
             .bind(version_hash)
-            .execute(&e.pool)
+            .execute(&mut *tx)
             .await
             .map_err(|x| {
                 error!("Error: {}", x);
                 MetadataStorageError::Internal
             })?;
 
-        // insert failed, todo proper error handling
-        if pg_result.rows_affected() != 1 {
-            return Err(MetadataStorageError::PackageDuplicate(
-                self.name.to_string(),
-                self.version.to_string(),
-            ));
+        let commit_result = tx.commit().await;
```

Comment:
Not quite right. Now you're doing exactly one query as part of the transaction, making it basically unnecessary. I line [140](https://github.com/helsing-ai/buffrs/pull/215/commits/05db820f49f395d60dc6c0be9abfc19d5ab59672#diff-b005606de71552e393b447fc882ecd3d9cfeedf4057e914209cf536cccc60507R140) you do a potential insert.

If `INSERT_VERSION_QUERY` fails, should we rollback `SELECT_OR_INSERT_PACKAGE_QUERY`? If yes, the transaction need to start before the latter and include both queries, if not, you can just remove the transaction all-together and go back to how this code looked before.